packages/contracts/contracts/MarginlyPool.sol

```diff
@@ -706,6 +684,9 @@ contract MarginlyPool is IMarginlyPool {
         require(realCollateralDelta <= quoteInMaximum, 'SL'); // Slippage above maximum
       }
 
+      realCollateralDelta = swapExactOutput(true, realQuoteCollateral, realBaseDebt, swapCalldata);
```

Comment:
Swap should be before slippage validation `require(realCollateralDelta <= quoteInMaximum, 'SL');`